### PR TITLE
[WIP] Automatically include web public streams

### PR DESCRIPTION
**Pull request by refeed**
Fixes https://github.com/zulip/zulip-archive/issues/80
Fixes https://github.com/zulip/zulip/issues/21789

### CODE REVIEW

**Comment by refeed** [comment]
Honestly, I am still confused, if we want to add all web public streams, we need to at least have a stream in included stream otherwise the script will just tell us to put `*` in there.

I wonder if it is okay to override that behavior, so it becomes: if there is no included_stream in streams.yaml, the script will just archive the web public streams implicitly instead rather than telling the user to put `*` in included_streams

**Comment by rht** [comment]
The python-zulip-api is hardcoded to 0.6.3: https://github.com/zulip/zulip-archive/blob/5da606feac0fa69445de717c406f79dd81e82e59/entrypoint.sh#L35. This needs to be updated so that `client.get_streams` accepts `include_web_public=True`.

**Comment by refeed** [comment]
@rht alright, I'm gonna update it to the latest version

File: lib/populate.py [hunk 1]
```diff
@@ -100,12 +100,9 @@ def safe_request(cmd, *args, **kwargs):
...
 def get_streams(client):
-    # In the future, we may want to change this to
-    # include_web_public=True, for organizations that might want to
-    # use the upcoming web_public flag; but at the very least we
-    # should only include public streams.
     response = safe_request(
-        client.get_streams, include_public=True, include_subscribed=False
+        client.get_streams, include_public=True, include_subscribed=False,
+        include_web_public=True
     )
```
Comment (timabbott) [inline lib/populate.py:106]: I think ideally we'd make this more of an option that can be controlled by users -- with a few settings in the configuration file for whether to include all web-public streams, all public streams, and maybe an option to list a specific set of streams.

File: entrypoint.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ python3 get-pip.py
 pip install virtualenv
 virtualenv -p python3 .
 source bin/activate
-pip3 install zulip==0.6.3
+pip3 install zulip==0.8.2
```
Comment (timabbott) [inline entrypoint.sh:35]: Let's do this a separate preparatory commit.

File: lib/populate.py (hunk 1 above)
Comment (rht) [reply lib/populate.py:106]: The public streams stuff can be done in a separate PR. I made #93 (which is finished, and currently I'm testing on julialang.zulipchat.com's web-public streams), so this PR can be closed.

**Comment by timabbott** [comment]
Closing in favor of #93.

**Pull request closed without merging**